### PR TITLE
Release/3.39.0

### PR DESCRIPTION
**Pull request by LetItCode**
Lets check release/3.39 merge in official repository. v3 community has been testing it for 4 months.

### CODE REVIEW

File: telegram.js [hunk 1]
```diff
@@ -54,17 +54,12 @@ class Telegram extends ApiClient {
...
   }
 
-  setWebhook (url, certificate, maxConnections, allowedUpdates) {
-    return this.callApi('setWebhook', {
-      url,
-      certificate,
-      max_connections: maxConnections,
-      allowed_updates: allowedUpdates
-    })
+  setWebhook (url, extra) {
```
Comment (Loskir) [inline telegram.js:57]: Breaking change

File: telegram.js [hunk 2]
```diff
@@ -316,14 +321,14 @@ class Telegram extends ApiClient {
     })
   }
 
-  editMessageLiveLocation (latitude, longitude, chatId, messageId, inlineMessageId, markup) {
+  editMessageLiveLocation (chatId, messageId, inlineMessageId, latitude, longitude, extra) {
```
Comment (Loskir) [inline telegram.js:324]: Breaking change

File: typings/telegraf.d.ts [hunk 3]
```diff
@@ -211,7 +206,13 @@ export interface TOptions {
    */
   channelMode?: boolean
 
+  /**
+   * Delay (in seconds) before making a follow-up request to get updates
+   */
   retryAfter?: number
 
+  /**
+   * Maximum amount of time (in microseconds) for which middlewares execution can pause updates fetching
```
Comment (Loskir) [inline typings/telegraf.d.ts:215]: Milliseconds?

**Review by Loskir — DISMISSED** [review]
There are some minor breaking changes, though rarely used. Discussion welcome

**Comment by wojpawlik** [comment]
I'll take a look when I'm back home at Monday.

I'd honestly rather v3 continued to live in letitcode/telegraf. What's the benefit of bringing it here? I don't think we can share code, so our issues will also be different.

**Comment by Loskir** [comment]
We want it published as 3.39.0

**Comment by wojpawlik** [comment]
Of course. I'm talking only about GitHub repository.

**Comment by LetItCode** [comment]
> Of course. I'm talking only about GitHub repository.

Yes we talk about it too. Live version must be here. I created fork for preparation to merge.

**Comment by wojpawlik** [comment]
Add support for Bot API 5.3, then I'll release 3.39 and discuss further. You can backport ff33055. What `dist-tag` do you want?

**Comment by Loskir** [comment]
We want 3.39 released now and then in 3.40 we will support the latest bot api and everything

**Comment by wojpawlik** [comment]
Fair enough.

> What `dist-tag` do you want?

`latest` is mine. `legacy` is the obvious choice. `classic` sounds less negative.

**Comment by wojpawlik** [comment]
Subsequent commits won't dismiss reviews anymore.

Docs need to be sorted: https://v4.telegrafjs.org is outdated somehow and thus should go. v3 docs should ideally be moved to https://v3.telegraf.js.org, I can redirect https://telegraf.js.org/v3 there.

**Review by abduraimovabdurahmon — APPROVED** [review]
Zor

**Comment by wojpawlik** [comment]
I've been thinking about stepping down. People who appreciate Telegraf 4 are moving to grammY. My [attempt](https://github.com/telegraf/telegraf/tree/deunionize) to fix #1388 is fragile.

Prove that you can continue maintaining 3.x. Add support for latest Bot API. Then I'll merge this, rename `develop` to `v4`, and grant @Loskir maintain and publish permissions.

**Comment by wojpawlik** [comment]
Perhaps @AndrewLaneX would be interested in maintaining Telegraf 4?

**Comment by vrumger** [comment]
Sorry, I've got a lot going on in my life right now and have also started moving some of my bots to grammY.

**Comment by wojpawlik** [comment]
IMPORTANT! https://github.com/telegraf/telegraf/discussions/1526

**Comment by wojpawlik** [comment]
Merging this broke https://telegraf.js.org/v3. PR welcome to replace the link in v4 Readme.

**Pull request merged**